online-shop/src/app/components/product-view/product-view.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, OnInit} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {ProductModel} from "../../types/product.model";
+import {HttpClient} from "@angular/common/http";
+import {ProductService} from "../../services/product.service";
+import {Location} from "@angular/common";
+import {UserService} from "../../services/user.service";
+
+@Component({
+  selector: 'app-product-view',
+  templateUrl: './product-view.component.html',
+  styleUrls: ['./product-view.component.scss']
+})
+export class ProductViewComponent implements OnInit {
+  product: ProductModel | undefined;
+  id: string | null = this.route.snapshot.paramMap.get('id');
```

Comment:
**suggestion:** you should have this assignment in the component lifecycle, more specifically in `ngOnInit`